### ISSUE TITLE
BibTeX file not found in resource path

### ISSUE THREAD

**Issue by rgaiacs** [quarto-dev/quarto-cli#860]
Related to https://github.com/quarto-dev/quarto-cli/discussions/853

# Minimal Working Example

Let's create a new book project:

```
$ quarto create-project mwe --type book
Creating project at /tmp/mwe:
  - Created _quarto.yml
  - Created index.qmd
  - Created intro.qmd
  - Created summary.qmd
  - Created references.qmd
  - Created cover.png
  - Created references.bib
$ cd mwe
$ quarto render --to html --output-dir public --debug --execute-debug --log-level info
[1/4] index.qmd
[2/4] intro.qmd
[3/4] summary.qmd
[4/4] references.qmd

Output created: public/index.html
```

`quarto` work as expected.

Let's create a directory called `foo` to host a part of our book and create a file `foo/index.qmd` to open our book part.

```
$ mkdir foo
$ cp index.qmd foo/
$ cat _quarto.yml 
project:
  type: book

book:
  title: "mwe"
  author: "Jane Doe"
  date: "13/05/2022"
  chapters:
    - index.qmd
    - intro.qmd
    - summary.qmd
    - references.qmd

bibliography: references.bib

format:
  html:
    theme: cosmo
  pdf:
    documentclass: scrreprt
$ quarto render --to html --output-dir public --debug --execute-debug --log-level info
[1/5] foo/index.qmd
[2/5] index.qmd
[3/5] intro.qmd
[4/5] summary.qmd
[5/5] references.qmd

File ../references.bib not found in resource path
ERROR: File ../references.bib not found in resource path
```

# Primary Issue

`quarto` fail with `File ../references.bib not found in resource path`. Some how, the introduction of `foo/index.qmd` messed with `quarto`'s path logic.

# Secondary Issue

`quarto` is processing `foo/index.qmd` but it is lot listed in `_quarto.yml`.

**Comment by dragonstyle** [comment]
Great find - I fixed two issues (incorrect bibliography path and incorrect rendering of files not in book) and a fresh should be available shortly. Thanks for letting us know about this!

**Comment by fuhrmanator** [comment]
I tried cloning my (working) project on a different Windows PC, with a newer update of quarto (0.9.410) and I think I hit this problem. On windows, it's a bit more obscure, probably because of the backslashes in the path (and I'm using git bash inside VSCode). It gives me this error:

    ERROR: File C:-ndc-quarto.bib not found in resource path
    
    C:\Users\me\Documents\GitHub\log210-ndc-quarto>

When I try `--render pdf` in this command, it gets even uglier (maybe because of the LaTeX chain?):

    Reverting Windows console CPs to (in,out) = (437,65001)
    ERROR: InvalidData: stream did not contain valid UTF-8

Just in case someone has these errors and searches.

**Comment by fuhrmanator** [comment]
The windows manifestation of not finding the bib resource I reported above is still present in 0.9.415. Here's what's in my _quarto.yml:

```yaml
book:
  title: "blah"
  author: 
    - "me"
    - "somebody else"
  subtitle: "![cover image](images/cover.png){width=70%}"
  date: "2022-05-06"
  chapters:
    - index.qmd
    - one.qmd
    - two.qmd
    - three.qmd
    - references.qmd

bibliography: references.bib
```

It turns out, the trouble is caused by the `.bib` and the `.qmd` having the same name (`references`). 

When I renamed `references.bib` to `my_bib.bib` and updated the yaml, the problem disappears (for both latex and html renders). I'm not sure this is the same bug, but it surely seems related to the logic of finding .bib files.
